spark/src/main/java/org/apache/iceberg/actions/RemoveOrphanFilesAction.java

```diff
@@ -272,13 +272,17 @@ private String metadataTableName(MetadataTableType type) {
 
       Predicate<FileStatus> predicate = file -> file.getModificationTime() < olderThanTimestamp;
 
-      int maxDepth = Integer.MAX_VALUE;
+      int maxDepth = 2000;
       int maxDirectSubDirs = Integer.MAX_VALUE;
 
       dirs.forEachRemaining(dir -> {
         listDirRecursively(dir, predicate, conf.value().value(), maxDepth, maxDirectSubDirs, subDirs, files);
       });
 
+      if (!subDirs.isEmpty()) {
+        throw new RuntimeException("Could not list dirs: " + subDirs);
```

Comment:
Can we use a better message here? I think this will only happen when the maximum depth is reached, so we could state that's what happened: `"Could not list location %s, reached maximum subdirectory depth %s", tableLocation, maxDepth`

Reply:
Updated. Unfortunately, we don't have `tableLocation` here as it is called for subdirs on executors.